delivery/common/jwt/lib/jwt.d.ts

```diff
@@ -0,0 +1,117 @@
+     * @param jwtOptions  Advanced options for validating JWT fields. see {@link JWTOptions} for more details.
+     * @returns           Instance of {@link JWTValidator}
+     * @throws {[Error](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error)} with appropriate error message if type check on {@link JWTOptions} fields fails.
+     * @throws {[DOMException](https://developer.mozilla.org/en-US/docs/Web/API/DOMException) | [TypeError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypeError)} when trying to use an invalid key data or when the key is not a key for the algorithm or when trying to use an algorithm that is either unknown or isn't suitable for a verify operation.
+     * @example Error('Invalid jwtOptions: issuer must be non empty string') - If type check fails for issuer field.
+     * @example Error('Invalid jwtOptions: clockTimestamp must be number') - If type check fails for clockTimestamp field.
+     */
+    constructor(jwtOptions?: JWTOptions);
+    /**
+     * Decodes the base64 url encoded token, applys JWT default rules and perform signature verification using @param keys.
+     * Default Rules are:
+     * - Token should have 2 or 3 parts. i.e header, payload and signature (if JWT is secured) in base64 url encoded.
+     * - If issuer verification enabled, matches issuer string with iss claim
+     * - If subject verification enabled, matches subject string with sub field
+     * - If audience verification enabled, matches audience string with aud field
+     * - If ignoreExpiration is false, validates JWT exp field
+     * - If ignoreNotBefore  is false, validates JWT nbf field
+     * @param base64JWTToken Base64 url encoded JWT token.
+     * @param keys           List of {@link CryptoKey } to be used for signature verification. Token is considered valid if signature is verifiable by any one key.
+     * @returns              Promise of {@link JWTJson}
+     * @throws {[Error](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error)} with appropriate error message if rules validation fails.
+     * @throws {[DOMException](https://developer.mozilla.org/en-US/docs/Web/API/DOMException) | [TypeError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypeError)} when trying to use an invalid key or when the key is not suitable for verify operation.
+     * @example Error('Invalid arguments!') - If argument type check fails.
+     * @example Error('JWT malformed: invalid jwt format') - If token is not in valid JWT format.
+     * @example Error('JWT token signature verification failed!') - If signature verification fails for all keys
+     */
+    validate(base64JWTToken: string, keys: CryptoKey[]): Promise<JWTJson>;
+    /**
+     * Validates data types for {@link JWTOptions} fields if present, else sets to default.
+     */
+    private validateOptionTypes;
```

Comment:
Doc style: Is it appropriate to include private functions in the API definition? I assume the TypeScript compiler won't let you call them, so I don't know if there's any value for the customer in providing Them.

Reply:
Hi @evan-hughes : we are using [dts()](https://www.npmjs.com/package/rollup-plugin-dts) plugin as of now to generate the TypeScript definition file. The plugin dont have any option to disable exporting private field. 
There is no value for customer, However it will be useful in our current pipeline to keep the files in edgeworker repo consistent with our internal repo. (manually changing the file in edgewokers would change the md5). 
May be we can keep it same for time being, i'll try to find a replacement for the plugin that does not export private fields.